htlcswitch/hop/iterator.go

```diff
@@ -181,8 +186,26 @@ func MakeBlindingKit(processor BlindingProcessor,
 				return nil, err
 			}
 
+			// If we have an override for the blinding point for
+			// the next node, we'll just use it without tweaking
+			// (the sender intended to switch out directly for
+			// this blinding point). Otherwise, we'll tweak our
+			// blinding point to get the next ephemeral key.
+			nextEph, err := processor.NextEphemeral(blinding)
```

Comment:
I think this is more logic that can be hidden behind a single intrerface that simply takes the encrypted blinded paths, and returns the next plaintext payload.

Reply:
Reading it at this site now, if you have the blinding factor, then you don't need any extra state to derive the next ephemeral, so this can just be a stand alone function. Still lean towards a single interface to properly invert the dependancies here. 

```go
type BlindingProcessor interface {
        ProcessBlindedHop(btcec.PublicKey, EncryptedBlindedPath) (*ForwardingInfo, error) 
}
```